homeassistant/components/solax/sensor.py

```diff
@@ -75,13 +108,23 @@ async def async_refresh(self, now=None):
 class Inverter(SensorEntity):
     """Class for a sensor."""
 
-    def __init__(self, uid, serial, key, unit):
+    def __init__(
+        self,
+        uid,
+        serial,
+        key,
+        unit,
+        state_class=None,
+        device_class=None,
+    ):
         """Initialize an inverter sensor."""
         self.uid = uid
         self.serial = serial
         self.key = key
         self.value = None
         self.unit = unit
+        self._state_class = state_class
+        self._device_class = device_class
```

Comment:
```suggestion
        self._attr_state_class = state_class
        self._attr_device_class = device_class
```

See: <https://developers.home-assistant.io/docs/core/entity#entity-class-or-instance-attributes>